### PR TITLE
Neque porro quisquam est qui dolorem

### PR DESCRIPTION
**Pull request by kkawula**
Closes #

## Introduced changes

-

## Checklist

- [ ] Linked relevant issue
- [ ] Updated relevant documentation
- [ ] Added relevant tests
- [ ] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/sncast/src/main.rs [hunk 1]
```diff
@@ -740,3 +737,49 @@ fn update_cast_config(config: &mut CastConfig, cli: &Cli) {
...
+
+    let config = load_global_config::<CastConfig>(&None, &cli.profile)?;
+
+    combine_cast_configs(&mut global_config, &config);
+
+    update_cast_config(&mut global_config, cli);
+    Ok(global_config)
+}
+
+fn combine_cast_configs(global_config: &mut CastConfig, config: &CastConfig) {
```
[the remark this answers is not in the document]
Comment (cptartur) [reply crates/sncast/src/main.rs:761]: With the current setup, if we added a new key to `CastConfig` we'd have to remember update this function. I'd rather have the compiler check this for us.

File: docs/src/projects/configuration.md [hunk 2]
```diff
@@ -105,6 +105,28 @@ response: [0x1, 0x23, 0x4]
...
+> 📝 **Note**
+> Global configuration file will be created automatically if it does not exist and contain predefined `default` profile.
+
+
+#### Interaction between Local and Global profiles
+
+Interaction between global and local config is based on the overriding mechanism, the local config will be used to override 
+the global profile of the same name or in case it is not defined to override the default global profile.
+
+If `--profile` is not provided, the default profile from the local (if present) and global configuration will be used.
```
Comment (cptartur) [inline docs/src/projects/configuration.md:128]: This needs to be written in simpler terms, so it's obvious what overrides what and in which cases.

File: CHANGELOG.md [hunk 3]
```diff
@@ -24,6 +24,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - You can skip `--name` flag when using `account import` - a default name will be generated.
 - Addresses outputted when calling `sncast account create`, `sncast deploy` and `sncast declare` are now padded to 64 characters length and prefixed with `0x0`
+- Globally available configuration to store profiles to share between projects.
+- Missing fields in `show-config` command.
```
Comment (ddoktorski) [inline CHANGELOG.md:28]: nit: Could you list the added fields here?

File: crates/shared/src/consts.rs [hunk 4]
```diff
@@ -1,3 +1,4 @@
 pub const EXPECTED_RPC_VERSION: &str = "0.7.0";
 pub const RPC_URL_VERSION: &str = "v0_7";
 pub const SNFORGE_TEST_FILTER: &str = "SNFORGE_TEST_FILTER";
+pub const FREE_RPC_PROVIDER_URL: &str = "https://free-rpc.nethermind.io/sepolia-juno/v0_7";
```
Comment (ddoktorski) [inline crates/shared/src/consts.rs:4]: nit: Do we really want to hardcode a link to an external node in the template? This might require updates in the future. I suggest sticking with the previous devnet address `http://127.0.0.1:5050`.

File: crates/sncast/src/helpers/global_config.rs [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    } else {
+        ".config/starknet-foundry"
+    });
+
+    if !global_config_dir.exists() {
+        fs::create_dir_all(&global_config_dir)?;
+    }
+
+    let global_config_path =
+        Utf8PathBuf::from_path_buf(global_config_dir.join("snfoundry.toml")).unwrap();
```
Comment (ddoktorski) [inline crates/sncast/src/helpers/global_config.rs:30]: nit: Better to use `expect` in such cases

File: crates/sncast/src/helpers/global_config.rs [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+    formatdoc! {r#"
+        # Visit https://foundry-rs.github.io/starknet-foundry/appendix/snfoundry-toml.html
+        # and https://foundry-rs.github.io/starknet-foundry/projects/configuration.html for more information
+
+        [sncast.default]
+        url = "{default_url}"
+        block-explorer = "{default_block_explorer}"
+        wait-params = {{ timeout = {default_wait_timeout}, retry-interval = {default_wait_retry_interval} }}
+        show-explorer-links = {default_show_explorer_links}
```
Comment (ddoktorski) [inline crates/sncast/src/helpers/global_config.rs:51]: Why this is not commented out?

File: crates/sncast/src/main.rs [hunk 7]
```diff
@@ -740,3 +737,49 @@ fn update_cast_config(config: &mut CastConfig, cli: &Cli) {
...
+    let global_config_path = get_global_config_path().unwrap_or_else(|err| {
+        eprintln!("Error getting global config path: {err}");
+        Utf8PathBuf::new()
+    });
+
+    let mut global_config =
+        load_global_config::<CastConfig>(&Some(global_config_path.clone()), &cli.profile)
+            .unwrap_or_else(|_| {
+                load_global_config::<CastConfig>(&Some(global_config_path), &None).unwrap()
+            });
```
Comment (ddoktorski) [inline crates/sncast/src/main.rs:751]: Do we need this part? If `profile` is not provided it should be `None`.

File: crates/sncast/src/main.rs [hunk 8]
```diff
@@ -740,3 +737,49 @@ fn update_cast_config(config: &mut CastConfig, cli: &Cli) {
         clone_or_else!(cli.wait_timeout, config.wait_params.get_timeout()),
     );
 }
+
+fn get_cast_config(cli: &Cli) -> Result<CastConfig> {
```
[the remark this answers is not in the document]
Comment (kkawula) [reply crates/sncast/src/main.rs:741]: The structure is called `CastConfig` so I suppose that it will be more accurate as it is now.

File: crates/sncast/src/helpers/global_config.rs [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+    let global_config_dir = config_dir.join(if cfg!(target_os = "windows") {
+        "starknet-foundry"
+    } else {
+        ".config/starknet-foundry"
+    });
+
+    if !global_config_dir.exists() {
+        fs::create_dir_all(&global_config_dir)?;
+    }
```
[the remark this answers is not in the document]
Comment (kkawula) [reply crates/sncast/src/helpers/global_config.rs:27]: Why not? I suppose that `.config` dir idea is intended to be used by tools, and we should create it if is not present

File: crates/shared/src/consts.rs (hunk 4 above)
Comment (kkawula) [reply crates/shared/src/consts.rs:4]: I suppose that it will be convenient for the user to have ready sepolia URL defined.

File: crates/sncast/src/helpers/configuration.rs [hunk 10]
```diff
@@ -57,7 +58,7 @@ impl Default for CastConfig {
             keystore: None,
             wait_params: ValidatedWaitParams::default(),
             block_explorer: Some(block_explorer::Service::default()),
-            show_explorer_links: true,
+            show_explorer_links: show_explorer_links_default(),
```
[the remark this answers is not in the document]
Comment (kkawula) [reply crates/sncast/src/helpers/configuration.rs:61]: Hmmmmmm, we are using it in a few places so maybe we should use const as it is now

File: crates/sncast/tests/e2e/main_tests.rs [hunk 11]
```diff
@@ -169,29 +169,6 @@ async fn test_missing_account_flag() {
...
-
-    let snapbox = runner(&args);
-    let output = snapbox.assert().failure();
-
-    assert_stderr_contains(
-        output,
-        "Error: RPC url not passed nor found in snfoundry.toml",
-    );
-}
-
```
[the remark this answers is not in the document]
Comment (kkawula) [reply crates/sncast/tests/e2e/main_tests.rs:194]: We provide it by default from now, so it isn't possible

File: crates/sncast/src/main.rs (hunk 8 above)
[the remark this answers is not in the document]
Comment (kkawula) [reply crates/sncast/src/main.rs:739]: It isn't possible at all, because we can export `Cli` to different module/file, refactor should be addressed in other issue

File: crates/sncast/src/main.rs [hunk 12]
```diff
@@ -740,3 +737,49 @@ fn update_cast_config(config: &mut CastConfig, cli: &Cli) {
...
+    let global_config_path = get_global_config_path().unwrap_or_else(|err| {
+        eprintln!("Error getting global config path: {err}");
+        Utf8PathBuf::new()
+    });
```
[the remark this answers is not in the document]
Comment (kkawula) [reply crates/sncast/src/main.rs:743]: discussed offline

File: docs/src/projects/configuration.md [hunk 13]
```diff
@@ -105,6 +105,28 @@ response: [0x1, 0x23, 0x4]
 </details>
 <br>
 
+### Global configuration
```
[the remark this answers is not in the document]
Comment (kkawula) [reply docs/src/projects/configuration.md:108]: It seems to me that the composition as it is now is quite natural. Check the changes. I hope it will fit to you

File: crates/sncast/src/helpers/global_config.rs (hunk 9 above)
[the remark this answers is not in the document]
Comment (cptartur) [reply crates/sncast/src/helpers/global_config.rs]: Seems bit weird to me to create those. 

But cc @THenry14 to verify

File: crates/sncast/src/main.rs (hunk 8 above)
[the remark this answers is not in the document]
Comment (cptartur) [reply crates/sncast/src/main.rs:739]: Let's create an issue to refactor that then. `main.rs` is to big at this stage.

File: crates/sncast/src/helpers/config.rs [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+}
+macro_rules! clone_field {
+    ($global_config:expr, $local_config:expr, $default_config:expr, $field:ident) => {
+        if $local_config.$field != $default_config.$field {
+            $local_config.$field.clone()
+        } else {
+            $global_config.$field.clone()
+        }
+    };
+}
```
Comment (cptartur) [inline crates/sncast/src/helpers/config.rs:79]: I like having new macros, however in this case I think it just could be a function :ccc

File: crates/shared/src/consts.rs (hunk 4 above)
Comment (ddoktorski) [reply crates/shared/src/consts.rs:4]: Although it might be convenient for some users I still don't think we should include it in the template. If you add it here you are kind of responsible for it to work, which means extra effort to maintain it and potentially dealing with related support questions.

File: crates/sncast/src/main.rs [hunk 15]
```diff
@@ -740,3 +735,23 @@ fn update_cast_config(config: &mut CastConfig, cli: &Cli) {
...
+        load_global_config::<CastConfig>(&Some(global_config_path.clone()), &cli.profile)
+            .unwrap_or_else(|_| {
+                load_global_config::<CastConfig>(&Some(global_config_path), &None).unwrap()
+            });
+
+    let local_config = load_global_config::<CastConfig>(&None, &cli.profile)?;
+
+    let mut combined_config = combine_cast_configs(&global_config, &local_config);
+
+    update_cast_config(&mut combined_config, cli);
```
Comment (cptartur) [inline crates/sncast/src/main.rs:755]: Can we rename this function? I know we had it before, but it's not a good name in my opinion for what it does.

File: docs/src/projects/configuration.md [hunk 16]
```diff
@@ -105,6 +105,56 @@ response: [0x1, 0x23, 0x4]
...
+- **A:** Global configuration file containing the profiles `default` and `testnet`.
+- **B:** Local configuration file containing the profiles `default` and `mainnet`.
+
+In any directory in the file system, a user can run the `sncast` command using the `default` and `testnet` profiles, 
+because they are located in global config. 
+If no profiles are explicitly specified, the `default` profile from the global configuration file will be used.
+
+When running `sncast` from the `opus magnum` directory, there is a configuration file in the parent directory (B). 
+This setup allows for the use of the following profiles: `default`, `testnet`, and `mainnet`. If the `mainnet` profile is specified, 
+the configuration from the local file will be used to override the global `default` profile, as the `mainnet` profile does not exist in the global configuration.
```
Comment (cptartur) [inline docs/src/projects/configuration.md:156]: It's a nice example but maybe show a shortened contents of A and B instead?

File: docs/src/projects/configuration.md [hunk 17]
```diff
@@ -105,6 +105,56 @@ response: [0x1, 0x23, 0x4]
 </details>
 <br>
 
+### Global configuration
+
+Global configuration file is usual [`snfoundry.toml`](https://foundry-rs.github.io/starknet-foundry/appendix/snfoundry-toml.html), 
+which is a common storage for configurations to apply to multiple projects across various directories.
+This file is stored in a predefined location and is managed differently compared to the local configuration file, which can be found in any parent directory.
+Profiles from global configurations are used to combine configuration with local profiles.
```
Comment (cptartur) [inline docs/src/projects/configuration.md:113]:
```suggestion
Global configuration file is usual [`snfoundry.toml`](https://foundry-rs.github.io/starknet-foundry/appendix/snfoundry-toml.html), 
which is a common storage for configurations to apply to multiple projects across various directories.
This file is stored in a predefined location and is used to store profiles that can be used from any location on your computer.
```

File: docs/src/projects/configuration.md [hunk 18]
```diff
@@ -105,6 +105,56 @@ response: [0x1, 0x23, 0x4]
...
+
+#### Interaction between Local and Global profiles
+
+Interaction between global and local config is based on the overriding mechanism
```
Comment (cptartur) [inline docs/src/projects/configuration.md:117]:
```suggestion
Global config can be overridden by a local config.
```

File: docs/src/projects/configuration.md [hunk 19]
```diff
@@ -105,6 +105,56 @@ response: [0x1, 0x23, 0x4]
...
+
+The local configuration will be used to override a global profile of the same name, or if no matching profile exists, it will override the global default profile. 
+If `--profile` is not explicitly specified, the default profiles from both the local and global configurations (if any is present) will be used.
```
Comment (cptartur) [inline docs/src/projects/configuration.md:120]:
```suggestion
If both local and global profiles with the same name are present, local profile will be combined with global profile. That is for any setting defined both in global and local profiles, local setting will be used. For settings not defined in local profile, value from the global profile will be used instead.

This same behavior applies for default profiles as well. A local default profile will override a global default profile.
```

Comment (cptartur) [reply docs/src/projects/configuration.md:120]: And link to a section above when talking about "default profiles".

**Review by cptartur — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 20]
```diff
@@ -24,6 +24,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - You can skip `--name` flag when using `account import` - a default name will be generated.
 - Addresses outputted when calling `sncast account create`, `sncast deploy` and `sncast declare` are now padded to 64 characters length and prefixed with `0x0`
+- Globally available configuration to store profiles to share between projects.
+- Missing fields in `show-config` command, `block_explorer` and `show_explorer_links`.
+- Default free sepolia testnet RPC URL for any `sncast` command.
```
Comment (ddoktorski) [inline CHANGELOG.md:29]: Even if we decide to add it, which I am strongly against, what is the point of mentioning this in the changelog?

**Review by ddoktorski — COMMENTED** [review]
In the commands `account add` and `account import` there is `--add-profile` flag. After introducing the global config it would be nice to explicitly mention that the created profile will be added to a local config if exists, or to change this logic. This can be done in this PR or a separate one, but it should be completed before the next release.

File: CHANGELOG.md (hunk 20 above)
Comment (kkawula) [reply CHANGELOG.md:29]: I suppose that it is a significant change. It changes the expected behavior from the older version. Isn't it proper to mention here?

File: crates/sncast/src/main.rs (hunk 8 above)
[the remark this answers is not in the document]
Comment (kkawula) [reply crates/sncast/src/main.rs:739]: #2700

File: CHANGELOG.md (hunk 20 above)
Comment (ddoktorski) [reply CHANGELOG.md:29]: I see now that this also applies to the default `CastConfig`. I thought it was only for the template. That makes it even worse, but in such a case it should be added to the changelog.

File: docs/src/projects/configuration.md (hunk 16 above)
Comment (kkawula) [reply docs/src/projects/configuration.md]: I've added links to proper sections, profile examples are defined above this section. I hope it will help avoid being redundant and also allow users to grab context. wdyt?

File: crates/sncast/src/helpers/config.rs [hunk 21]
```diff
@@ -0,0 +1,114 @@
...
+        # url = "{default_url}"
+        # block-explorer = "{default_block_explorer}"
+        # wait-params = {{ timeout = {default_wait_timeout}, retry-interval = {default_wait_retry_interval} }}
+        # show-explorer-links = {default_show_explorer_links}
+        # accounts-file = "{default_accounts_file}"
+        # account = "{default_account}"
+        # keystore = "{default_keystore}"
+        "#,
+        default_url = FREE_RPC_PROVIDER_URL,
+        default_accounts_file = "~/.starknet_accounts/starknet_open_zeppelin_accounts.json",
```
Comment (ddoktorski) [inline crates/sncast/src/helpers/config.rs:55]: nit: There is a const `DEFAULT_ACCOUNTS_FILE`

File: crates/shared/src/consts.rs (hunk 4 above)
Comment (ddoktorski) [inline crates/shared/src/consts.rs:4]: nit: Should be removed completely for now

File: crates/forge/src/init.rs [hunk 22]
```diff
@@ -20,16 +21,19 @@ fn create_snfoundry_manifest(path: &PathBuf) -> Result<()> {
...
+
+        # [sncast.default]                                         # Define a profile name
+        # url = "{default_rpc_url}" # Url of the RPC provider
+        # accounts-file = "../account-file"                        # Path to the file with the account data
+        # account = "mainuser"                                     # Account from `accounts_file` or default account file that will be used for the transactions
+        # keystore = "~/keystore"                                  # Path to the keystore file
+        # wait-params = {{ timeout = 300, retry-interval = 10 }}     # Wait for submitted transaction parameters
+        # block-explorer = "StarkScan"                             # Block explorer service used to display links to transaction details
+        # show-explorer-links = true                               # Print links pointing to pages with transaction details in the chosen block explorer
+        "#,
```
Comment (ddoktorski) [inline crates/forge/src/init.rs:35]: nit: Why not to share the default template between init and global config?

File: crates/sncast/src/main.rs [hunk 23]
```diff
@@ -740,3 +735,23 @@ fn update_cast_config(config: &mut CastConfig, cli: &Cli) {
...
+        Utf8PathBuf::new()
+    });
+
+    let global_config =
+        load_global_config::<CastConfig>(&Some(global_config_path.clone()), &cli.profile)
+            .unwrap_or_else(|_| {
+                load_global_config::<CastConfig>(&Some(global_config_path), &None).unwrap()
+            });
+
+    let local_config = load_global_config::<CastConfig>(&None, &cli.profile)?;
```
Comment (cptartur) [inline crates/sncast/src/main.rs:751]: It's loading local config. Please rename this function.

File: crates/sncast/src/main.rs [hunk 24]
```diff
@@ -740,3 +735,23 @@ fn update_cast_config(config: &mut CastConfig, cli: &Cli) {
...
+
+    let mut combined_config = combine_cast_configs(&global_config, &local_config);
+
+    update_config_with_cli(&mut combined_config, cli);
```
Comment (cptartur) [inline crates/sncast/src/main.rs:755]: Can we change the function `update_config_with_cli` to something like `config_with_cli` or similar and make it return a new config instead of taking a mutable reference?

File: docs/src/projects/configuration.md [hunk 25]
```diff
@@ -105,6 +105,56 @@ response: [0x1, 0x23, 0x4]
 </details>
 <br>
 
+### Global configuration
+
+Global configuration file is usual [`snfoundry.toml`](https://foundry-rs.github.io/starknet-foundry/appendix/snfoundry-toml.html),
```
Comment (cptartur) [inline docs/src/projects/configuration.md:110]:
```suggestion
Global configuration file is a [`snfoundry.toml`](https://foundry-rs.github.io/starknet-foundry/appendix/snfoundry-toml.html), 
```

File: docs/src/projects/configuration.md [hunk 26]
```diff
@@ -105,6 +105,56 @@ response: [0x1, 0x23, 0x4]
...
+
+Global configuration file is usual [`snfoundry.toml`](https://foundry-rs.github.io/starknet-foundry/appendix/snfoundry-toml.html), 
+which is a common storage for configurations to apply to multiple projects across various directories.
+This file is stored in a predefined location and is used to store profiles that can be used from any location on your computer.
+
+#### Interaction between Local and Global profiles
+
+Global config can be overridden by a local config.
+
+If both local and global profiles with the same name are present, local profile will be combined with global profile. That is for any setting defined both in global and local profiles, local setting will be used. For settings not defined in local profile, value from the global profile will be used instead.
```
Comment (cptartur) [inline docs/src/projects/configuration.md:118]:
```suggestion
If both local and global profiles with the same name are present, local profile will be combined with global profile. That is for any setting defined both in global and local profiles, local setting will be used. For settings not defined in local profile, values from the global profile will be used instead.
```

File: docs/src/projects/configuration.md [hunk 27]
```diff
@@ -105,6 +105,56 @@ response: [0x1, 0x23, 0x4]
...
+#### Global Configuration File Location
+The global configuration is stored in a specific location depending on the operating system:
+
+- macOS/Linux : The global configuration file is located at `$HOME/.config/starknet-foundry/snfoundry.toml`
+- Windows : The file can be found at `C:\Users\<user>\AppData\Roaming\starknet-foundry\snfoundry.toml`
+
+> 📝 **Note**
+> If missing, global configuration file will be created automatically on running any `sncast` command for the first time.
+
+### Config interaction example
```
Comment (cptartur) [inline docs/src/projects/configuration.md:131]:
```suggestion
### Config Interaction Example
```

File: docs/src/projects/configuration.md [hunk 28]
```diff
@@ -105,6 +105,56 @@ response: [0x1, 0x23, 0x4]
 </details>
 <br>
 
+### Global configuration
+
+Global configuration file is usual [`snfoundry.toml`](https://foundry-rs.github.io/starknet-foundry/appendix/snfoundry-toml.html), 
+which is a common storage for configurations to apply to multiple projects across various directories.
+This file is stored in a predefined location and is used to store profiles that can be used from any location on your computer.
+
+#### Interaction between Local and Global profiles
```
Comment (cptartur) [inline docs/src/projects/configuration.md:114]:
```suggestion
#### Interaction Between Local and Global Profiles
```

File: docs/src/projects/configuration.md [hunk 29]
```diff
@@ -105,6 +105,56 @@ response: [0x1, 0x23, 0x4]
 </details>
 <br>
 
+### Global configuration
```
Comment (cptartur) [inline docs/src/projects/configuration.md:108]:
```suggestion
### Global Configuration
```

File: docs/src/projects/configuration.md [hunk 30]
```diff
@@ -105,6 +105,56 @@ response: [0x1, 0x23, 0x4]
...
+
+In any directory in the file system, a user can run the `sncast` command using the `default` and `testnet` profiles, 
+because they are located in global config. 
+If no profiles are explicitly specified, the `default` profile from the global configuration file will be used.
+
+When running `sncast` from the `opus-magnum` directory, there is a configuration file in the parent directory (B). 
+This setup allows for the use of the following profiles: `default`, `testnet`, and `mainnet`. If the `mainnet` profile is specified, 
+the configuration from the local file will be used to override the global `default` profile, as the `mainnet` profile does not exist in the global configuration.
+
 ## Environmental variables
```
Comment (cptartur) [inline docs/src/projects/configuration.md:158]:
```suggestion
## Environmental Variables
```

File: docs/src/projects/configuration.md [hunk 31]
```diff
@@ -105,6 +105,56 @@ response: [0x1, 0x23, 0x4]
...
+                └── opus-magnum/
+```
+
+**Glossary:**
+
+- **A:** Global configuration file containing the profiles [`default`](#default-profile) and [`testnet`](#defining-profiles-in-snfoundrytoml).
+- **B:** Local configuration file containing the profiles `default` and `mainnet`.
+
+In any directory in the file system, a user can run the `sncast` command using the `default` and `testnet` profiles, 
+because they are located in global config.
```
Comment (cptartur) [inline docs/src/projects/configuration.md:151]:
```suggestion
In any directory in the file system, a user can run the `sncast` command using the `default` and `testnet` profiles, 
because they are defined in global config (file A). 

```

File: docs/src/projects/configuration.md [hunk 32]
```diff
@@ -105,6 +105,56 @@ response: [0x1, 0x23, 0x4]
...
+**Glossary:**
+
+- **A:** Global configuration file containing the profiles [`default`](#default-profile) and [`testnet`](#defining-profiles-in-snfoundrytoml).
+- **B:** Local configuration file containing the profiles `default` and `mainnet`.
+
+In any directory in the file system, a user can run the `sncast` command using the `default` and `testnet` profiles, 
+because they are located in global config. 
+If no profiles are explicitly specified, the `default` profile from the global configuration file will be used.
+
+When running `sncast` from the `opus-magnum` directory, there is a configuration file in the parent directory (B).
```
Comment (cptartur) [inline docs/src/projects/configuration.md:154]:
```suggestion
When running `sncast` from the `opus-magnum` directory, there is a configuration file in the parent directory (file B). 
```

File: docs/src/projects/configuration.md [hunk 33]
```diff
@@ -105,6 +105,56 @@ response: [0x1, 0x23, 0x4]
...
+This file is stored in a predefined location and is used to store profiles that can be used from any location on your computer.
+
+#### Interaction between Local and Global profiles
+
+Global config can be overridden by a local config.
+
+If both local and global profiles with the same name are present, local profile will be combined with global profile. That is for any setting defined both in global and local profiles, local setting will be used. For settings not defined in local profile, value from the global profile will be used instead.
+
+This same behavior applies for [default profiles](#default-profile) as well. A local default profile will override a global default profile.
+
```
Comment (ddoktorski) [inline docs/src/projects/configuration.md:121]: I think the information that CLI values will always override values from both configs should be added somewhere

File: crates/sncast/src/helpers/configuration.rs (hunk 10 above)
[the remark this answers is not in the document]
Comment (ddoktorski) [reply crates/sncast/src/helpers/configuration.rs:60]: What was decided here? Can we create a const in the `constants.rs` file instead?

File: crates/sncast/src/main.rs (hunk 23 above)
Comment (kkawula) [reply crates/sncast/src/main.rs]: Renamed. I've also renamed `GlobalConfig` trait. This is no longer the correct name

File: crates/shared/src/consts.rs (hunk 4 above)
Comment (kkawula) [reply crates/shared/src/consts.rs:4]: Yes, it was removed as a default from the CastConfig::Default, but I would like to leave it in the commented template.

**Review by cptartur — APPROVED** [review]

File: docs/src/projects/configuration.md [hunk 34]
```diff
@@ -105,7 +105,62 @@ response: [0x1, 0x23, 0x4]
...
+Global configuration file is a [`snfoundry.toml`](https://foundry-rs.github.io/starknet-foundry/appendix/snfoundry-toml.html), 
+which is a common storage for configurations to apply to multiple projects across various directories.
+This file is stored in a predefined location and is used to store profiles that can be used from any location on your computer.
+
+#### Interaction Between Local and Global Profiles
+
+Global config can be overridden by a local config.
+
+If both local and global profiles with the same name are present, local profile will be combined with global profile. That is for any setting defined both in global and local profiles, local setting will be used. For settings not defined in local profile, values from the global profile will be used instead.
```
Comment (ddoktorski) [inline docs/src/projects/configuration.md:118]: The logic of overriding global default profil should be explicitly mention here, not only in the example below
```suggestion
If both local and global profiles with the same name are present, local profile will be combined with global profile. For any setting defined in both profiles, the local setting will take precedence. For settings not defined in the local profile, values from the corresponding global profile will be used, or if not defined, values from the global default profile will be used instead.
```

**Review by ddoktorski — APPROVED** [review]
